Miri: Renamed "undef" to "uninit"

pnadon opened the pull request:
Renamed remaining references to "undef" to "uninit" when referring to Miri.

Impacted directories are:

- `src/librustc_codegen_llvm/consts.rs`
- `src/librustc_middle/mir/interpret/`
- `src/librustc_middle/ty/print/pretty.rs`
- `src/librustc_mir/`
- `src/tools/clippy/clippy_lints/src/consts.rs`

Upon building Miri based on the new changes it was verified that no changes needed to be made with the Miri project.

Related issue #71193

rust-highfive commented:
r? @eddyb

(rust_highfive has picked a reviewer for you, use r? to override)

pnadon commented:
The test initially failed due to formatting on a particular line in `src/librustc_mir/util/pretty.rs`. I ran the test which failed on my local machine, and it seems to be ok now.

RalfJung commented:
r? @RalfJung

RalfJung commented:
(Next time please Cc me in the PR description or put a link to the PR in the issue when there was prior discussion; I almost missed that you created this PR.)

pnadon commented:
> (Next time please Cc me in the PR description or put a link to the PR in the issue when there was prior discussion; I almost missed that you created this PR.)

Ok sounds good, I was thinking of doing so but did not want to spam you. I will do so next time!

bors commented:
:umbrella: The latest upstream changes (presumably #74821) made this pull request unmergeable. Please resolve the merge conflicts.

RalfJung commented:
Please don't add merge commits to PRs; instead, please rebase on top of current master.

RalfJung commented on src/librustc_middle/mir/interpret/allocation.rs:
```diff
@@ -636,8 +636,8 @@ impl<Tag, Extra> Allocation<Tag, Extra> {
     ) {
-        // An optimization where we can just overwrite an entire range of definedness bits if
-        // they are going to be uniformly `1` or `0`.
+        // An optimization where we can just overwrite an entire range of an entire range of
+        // initialization bits bits if they are going to be uniformly `1` or `0`.
```
```suggestion
        // An optimization where we can just overwrite an entire range of
        // initialization bits if they are going to be uniformly `1` or `0`.
```

Commit:
Miri: Renamed "undef" to "uninit"

Renamed remaining references to "undef" to "uninit" when referring to Miri.

Impacted directories are:

- src/librustc_codegen_llvm/consts.rs
- src/librustc_middle/mir/interpret/
- src/librustc_middle/ty/print/pretty.rs
- src/librustc_mir/
- src/tools/clippy/clippy_lints/src/consts.rs

Upon building Miri based on the new changes it was verified that no changes needed to be made with the Miri project.

Related issue #71193

RalfJung commented:
@bors r+ rollup
(there's no email notification after a force-push so I did not realize this was ready for review again. feel free to ping me in such cases.)

bors commented:
:pushpin: Commit 0c6e8b61b6a4df87216570931c687874f28022fd has been approved by `RalfJung`

<!-- @bors r=RalfJung 0c6e8b61b6a4df87216570931c687874f28022fd -->
<!-- homu: {"type":"Approved","sha":"0c6e8b61b6a4df87216570931c687874f28022fd","approver":"RalfJung"} -->
